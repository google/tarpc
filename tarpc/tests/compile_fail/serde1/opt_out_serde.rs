#![allow(deprecated)]

use std::fmt::Formatter;

#[tarpc::service(derive_serde = false)]
trait Foo {
    async fn foo();
}

fn foo(f: &mut Formatter) {
    let x = FooRequest::Foo {};
    tarpc::serde::Serialize::serialize(&x, f);
}

fn main() {}
