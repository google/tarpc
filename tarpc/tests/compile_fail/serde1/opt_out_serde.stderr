error[E0277]: the trait bound `FooRequest: Serialize` is not satisfied
  --> tests/compile_fail/serde1/opt_out_serde.rs:12:40
   |
12 |     tarpc::serde::Serialize::serialize(&x, f);
   |     ---------------------------------- ^^ the trait `Serialize` is not implemented for `FooRequest`
   |     |
   |     required by a bound introduced by this call
   |
   = note: for local types consider adding `#[derive(serde::Serialize)]` to your `FooRequest` type
   = note: for types from other crates check whether the crate offers a `serde` feature flag
   = help: the following other types implement trait `Serialize`:
             &'a T
             &'a mut T
             ()
             (T,)
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
           and $N others
